sktime/forecasting/base/_base.py

```diff
@@ -23,10 +59,31 @@ class BaseForecaster(BaseEstimator):
 
     def __init__(self):
         self._is_fitted = False
+
+        self._y = None
+        self._X = None
+
+        # forecasting horizon
+        self._fh = None
+        self._cutoff = None  # reference point for relative fh
+
+        # defaults for estimator tags
+        if not hasattr(self, "_tags"):
```

Comment:
Can we handle tags on the class level rather than object level? See the tag functionality on the `BaseEstimator`

Reply:
this is necessary to make the mixin workaround work, to ensure compatibility with the current interface. I would argue to refactor this later and accept a temporary inconsistency in the `_tags` interface, to allow dependent code to work.

Mid-term, I agree that tags should all live in class variables.

Reply:
But we have tags working already, so why not use that? Which part didn't work with the current system?

Reply:
Yes, using tags for "fh needed in fit" would work, and that is my long-term goal.

But I did not want to make invasive changes to existing forecasters, so I built a workaround that the tag gets created on creation when something inherits from the mixin. Long-term, I´d want to remove the mixins, and work with tags only.

Short-term, I decided to make mixins work like that so we can first do the interface refactor and then slowly remove the "patches" that make it work with the current inheritance structure. That would see the mixins and `_SktimeForecaster` fully removed - but not in this PR to avoid the "too many files" problem.

Reply:
see #950